tests/test_backend.py

```diff
@@ -1117,6 +1122,37 @@ def test_actual_expiry(self):
 
 
 class TestDefaultClient(unittest.TestCase):
+    def setUp(self):
+        self.client = caches[DEFAULT_CACHE_ALIAS].client
+
+    def tearDown(self):
+        try:
+            self.client.clear()
+        except exceptions.ConnectionInterrupted:
+            pass
+        finally:
+            self.client = None
+
+    def test_close_client_disconnect_default(self):
+        self.client.get_client(write=True)
+        with patch("redis.connection.ConnectionPool.disconnect") as mock:
+            self.client.close()
+            assert mock.called
+
+    @override_settings(DJANGO_REDIS_CLOSE_CONNECTION=False)
+    def test_close_disconnect_settings(self):
+        self.client.get_client(write=True)
+        with patch("redis.connection.ConnectionPool.disconnect") as mock:
```

Comment:
I do not know if this works with other clients

Reply:
Correct and agree: solved with the definition of disconnect method on connection_factory (if that implementation makes sense)

Reply:
makes sense, nice catch. Because also the connect is delegated to the connection factory